velox/exec/fuzzer/WindowFuzzer.cpp

```diff
@@ -89,19 +89,30 @@ std::tuple<std::string, bool> WindowFuzzer::generateFrameClause() {
   }
 
   auto frameBound =
-      [minKValue, maxKValue, this](
-          core::WindowNode::BoundType boundType) -> const std::string {
-    // Generating only constant bounded k PRECEDING/FOLLOWING frames for now.
+      [minKValue, maxKValue, this, &argTypes, &argNames](
+          core::WindowNode::BoundType boundType, std::string colVal) -> const std::string {
     auto kValue = boost::random::uniform_int_distribution<int64_t>(
         minKValue, maxKValue)(rng_);
     switch (boundType) {
       case core::WindowNode::BoundType::kUnboundedPreceding:
         return "UNBOUNDED PRECEDING";
       case core::WindowNode::BoundType::kPreceding:
+        if (vectorFuzzer_.coinToss(0.5)) {
+          // Generating column bounded k PRECEDING frames 50% of the time and constant bounded the other 50%
+          argTypes.push_back(INTEGER());
```

Comment:
Think the frame values in the column need to be non-negative. Please can you check the generation. Else we would have many failing tests.

Reply:
+1. It looks like `boost::random::uniform_int_distribution` in `VectorFuzzer::rand()` has 0 for default value of min_arg so negative values are likely not generated. Could you please confirm @jkhaliqi ?

Reply:
Right as @pramodsatya mentioned from the code the min argument value is 0 so negative values are not generated...Also looking through and pasting LOG(INFO) for the values generated and not seeing negatives being generated as well, but 0 being the minimum number present after a while so assuming no negatives will show up.

Reply:
Thanks @jkhaliqi for confirming. Please can you add a comment for it as well.

Reply:
Thank you, Added comment to the code stating how values are still positive

Reply:
Hi @jkhaliqi and @aditi-pandit, shall we generate negative values occasionally just to ensure the Window operator rejects them as expected? We can do this in a separate PR if you agree.

Reply:
@kagamiori : We could always do exactly one test case that checks the error with a randomly generated negative number in the fuzzer. Another option is to change the invalidFrames test to use a randomly generated -ve value in its logic https://github.com/facebookincubator/velox/blob/main/velox/functions/prestosql/window/tests/NthValueTest.cpp#L222

Are we doing such negative cases as a policy in the fuzzer ? If yes, then I'm fine else the unit test case is good as well.

Reply:
@aditi-pandit Good point. I agree that unit test is a more appropriate option for the test coverage in this case.